### PR TITLE
JavaScript formatting between `script` tags

### PR DESCRIPTION
**Pull request by PolinaShneider**
JavaScript formatting inside `script` tags via [esprima](https://esprima.org/) & [escodegen](https://github.com/estools/escodegen/wiki/API)

formatting_upd

### CODE REVIEW

**Comment by annstriganova** [comment]
There must be 2 spaces indent inside JS statement block:
Selection_001

File: snippets/snippets.json [hunk 1]
```diff
@@ -665,8 +665,8 @@
         "body": [
             "script",
             "  window.${1:userFunction} = function () {",
-            "  ${2:return Math.round(value / 10) * 10;}",
-            "  }",
+            "    ${2:return Math.round(value / 10) * 10;}",
```
Comment (VeselovAlex) [inline snippets/snippets.json]: I think `  ${2}` with indent is enough

File: server/src/formatter.ts [hunk 2]
```diff
@@ -111,6 +118,45 @@ export class Formatter {
...
+            format: {
+                indent: {
+                    base: (this.currentIndent.length / this.options.tabSize) + 1,
+                    style: " ".repeat(this.options.tabSize)
+                }
+            }
+        });
+
+        // If formatted code is the same as original, no need to make edit
+        if (formattedCode === content) {
```
Comment (VeselovAlex) [inline server/src/formatter.ts]: I think this check is unnecessary. Will it really improve performance or it is just wastes time for string comparision?

**Comment by VeselovAlex** [comment]
Move `package.json` fixes to separate branch and PR. You can use `git cherry-pick` to perform commit move.

**Pull request merged**